traits/has_traits.py

```diff
@@ -1161,7 +1188,7 @@ def _add_class_trait(cls, name, trait, is_subclass):
             prefix_list.append(name)
 
             # Resort the list from longest to shortest:
-            prefix_list.sort(lambda x, y: len(y) - len(x))
+            prefix_list.sort(key=lambda x: -len(x))
```

Comment:
Perhaps?
```suggestion
            prefix_list.sort(key=lambda x: len(x), reverse=True)
```

Reply:
Yep, that works too, but I think I'd rather keep this consistent with the code here: https://github.com/enthought/traits/blob/d8d4e8ffb502e31375e5706b2b989ad166839571/traits/has_traits.py#L646

(We could change both, of course.)

Reply:
> (We could change both, of course.)

Done.